File: backend/src/main/java/com/NBE3_4_2_Team4/domain/base/genFile/dto/GenFileDto.java
package com.NBE3_4_2_Team4.domain.base.genFile.dto;

import com.NBE3_4_2_Team4.domain.base.genFile.entity.GenFile;
import lombok.Getter;
import org.springframework.lang.NonNull;

import java.time.LocalDateTime;

@Getter
public class GenFileDto<T extends GenFile> {
    @NonNull
    private long id;
    @NonNull
    private LocalDateTime createdAt;
    @NonNull
    private long parentId;
    @NonNull
    private String fileName;
    @NonNull
    private GenFile.TypeCode typeCode;
    @NonNull
    private String fileExtTypeCode;
    @NonNull
    private String fileExtType2Code;
    @NonNull
    private long fileSize;
    @NonNull
    private long fileNo;
    @NonNull
    private String fileExt;
    @NonNull
    private String fileDateDir;
    @NonNull
    private String originalFileName;
    @NonNull
    private String downloadUrl;
    @NonNull
    private String publicUrl;

    public GenFileDto(T genFile) {
        this.id = genFile.getId();
        this.createdAt = genFile.getCreatedAt();
        this.parentId = genFile.getParent().getId();
        this.fileName = genFile.getFileName();
        this.typeCode = genFile.getTypeCode();
        this.fileExtTypeCode = genFile.getFileExtTypeCode();
        this.fileExtType2Code = genFile.getFileExtType2Code();
        this.fileSize = genFile.getFileSize();
        this.fileNo = genFile.getFileNo();
        this.fileExt = genFile.getFileExt();
        this.fileDateDir = genFile.getFileDateDir();
        this.originalFileName = genFile.getOriginalFileName();
        this.downloadUrl = genFile.getDownloadUrl();
        this.publicUrl = genFile.getPublicUrl();
    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/base/genFile/dto/GenFileDto.kt
package com.NBE3_4_2_Team4.domain.base.genFile.dto;

import com.NBE3_4_2_Team4.domain.base.genFile.entity.GenFile
import java.time.LocalDateTime

class GenFileDto<T : GenFile<*>>(
    val id: Long,
    val createdAt: LocalDateTime,
    val parentId: Long,
    val fileName: String,
    val typeCode: GenFile.TypeCode,
    val fileExtTypeCode: String,
    val fileExtType2Code: String,
    val fileSize: Int,
    val fileNo: Int,
    val fileExt: String,
    val fileDateDir: String,
    val originalFileName: String,
    val downloadUrl: String,
    val publicUrl: String,
    val fileType: String
) {
    constructor(genFile: T) : this(
        id = genFile.id,
        createdAt = genFile.createdAt,
        parentId = genFile.parent.id,
        fileName = genFile.fileName,
        typeCode = genFile.typeCode,
        fileExtTypeCode = genFile.fileExtTypeCode,
        fileExtType2Code = genFile.fileExtType2Code,
        fileSize = genFile.fileSize,
        fileNo = genFile.fileNo,
        fileExt = genFile.fileExt,
        fileDateDir = genFile.fileDateDir,
        originalFileName = genFile.originalFileName,
        downloadUrl = genFile.downloadUrl,
        publicUrl = genFile.publicUrl,
        fileType = genFile.modelName.substring(0, genFile.modelName.indexOf("GenFile"))
    )
}
